File: problem_16/power_digit_sum_test.rb
require 'minitest/autorun'
require 'minitest/pride'

class PowerDigitSumTest < Minitest::Test
  def power_digit_sum(base, exp)
    num = (base ** exp).to_s
    sum = 0

    num.chars.each { |n| sum += n.to_i }

    sum
  end

  #----------------- Tests -----------------#

  def test_power_digit_sum
    assert_equal 26,    power_digit_sum(2, 15)
    assert_equal 18,    power_digit_sum(3, 6)
    assert_equal 1366,  power_digit_sum(2, 1000)
  end
end

File: problem_16/power-digit-sum.test.js
function powerDigitSum(base, exp) {
  let numString = BigInt(Math.pow(base, exp)).toString();
  let sum = 0;

  for (i = 0; i < numString.length; i++) {
    sum += parseInt(numString[i]);
  }

  return sum;
}

test("powerDigitSum", () => {
  expect(powerDigitSum(2, 15)).toBe(26);
  expect(powerDigitSum(3, 6)).toBe(18);
  expect(powerDigitSum(2, 1000)).toBe(1366);
});
